### PR TITLE
Make Products.Archetypes dependency optional

### PR DESCRIPTION
**Pull request by jaroel**
Implements #197

### CODE REVIEW

File: src/plone/api/content.py [hunk 1]
```diff
@@ -18,9 +17,19 @@
...
 
+import pkg_resources
 import random
 import transaction
 
+try:
+    pkg_resources.get_distribution('Products.Archetypes')
+except pkg_resources.DistributionNotFound:
+    class IBaseObject(Interface):
+        """ Products.Archetypes.interfaces.base.IBaseObject
```
Comment (zupo) [inline src/plone/api/content.py]: Not according to docstring conventions. Should be a single line (if short enough), no leading/trailing spaces.

**Comment by zupo** [comment]
One minor styling nitpick, otherwise looks good to me!

**Comment by jaroel** [comment]
Please review.

**Comment by mattss** [comment]
Travis build failed due to a network error. I've restarted it.

**Comment by jaroel** [comment]
Yeah, a merge. Somehow I couldn't rebase.

**Comment by jaroel** [comment]
Please review.

File: src/plone/api/tests/test_content.py [hunk 2]
```diff
@@ -183,6 +190,10 @@ def test_create_dexterity(self):
                 id='test-item',
             )
 
+    @unittest.skipUnless(
+        HAS_ARCHETYPES,
+        "Only run when Dexterity is available.",
```
Comment (hvelarde) [inline src/plone/api/tests/test_content.py]: "Only run when **Archetypes** is available."

Comment (jaroel) [inline src/plone/api/tests/test_content.py]: Fixed, thanks!

**Pull request merged**